common/arm.h

```diff
@@ -0,0 +1,18 @@
+#ifndef _ARM_H
+#define _ARM_H
+
+#ifndef __ASSEMBLY__
+
+#if __ARM_ARM < 6
+static inline void wfi(void)
+{
+	asm volatile (
+		"mov r0, #0\n\t"
+		"mcr p15, 0, r0, c7, c0, 4\n\t"::: "r0", "cc", "memory");
+}
+#else
+static inline void wfi(void) { asm volatile ("wfi":::"memory"); }
```

Comment:
as noticed on the stream, this preprocessor guard might not be working as intended.

Reply:
Ah, should be `__ARM_ARCH`.